app.js

```diff
@@ -0,0 +1,170 @@
+
+// set the port for the server
+const PORT = 4200
+
+// load the required modules
+const http = require('http'),
+	express = require('express'),
+	cors = require('cors'),
+	bodyParser = require('body-parser');
+
+// create the server
+const app = express();
+const server = http.createServer(app);
+
+// configure the server
+const corsOptions = {
+	methods: ['GET', 'PUT', 'POST'],
+	origin: '*',
+	allowedHeaders: ['Content-Type', 'Accept', 'X-Requested-With', 'Origin']
+};
+
+app.use(cors(corsOptions));
+app.use(bodyParser.json());
+
+//connect to the database and store the reference
+const MongoClient = require('mongodb').MongoClient;
+var db;
+
+// Get the objectID type
+var ObjectID = require('mongodb').ObjectID;
+
+// connect to the database
+MongoClient.connect('mongodb://localhost:4201/tutorial-builder', (err, database) => {
+  if (err) {
+    throw err;
+  }
+  db = database;
+
+	// then start the server
+	server.listen(PORT);
+	console.log('Listening on port %d', server.address().port);
+});
+
+// setup the routes
+app.get('/', (req, res)=>{
+	res.send("<h1>Homepage</h1>");
+});
+
+// get the info about the tutorials
+app.get('/tutorials/:id*?', (req, res) => {
```

Comment:
What do you think about using the mongo drivers ES6 features (promises), seen here: https://mongodb.github.io/node-mongodb-native/2.2/tutorials/crud/
There's a little slider on the site to switch from Javascript 5 to JS6, so make sure you toggle that to see what I'm talking about.